examples/hybrid/callbacks.jl

```diff
@@ -189,40 +191,90 @@ function save_to_disk_func(integrator)
 
     # cloudwater (liquid and ice), watervapor, precipitation, and RH for moist simulation
     if :ρq_tot in propertynames(Y.c)
+
         ᶜq = @. TD.PhasePartition(thermo_params, ᶜts)
         ᶜcloud_liquid = @. ᶜq.liq
         ᶜcloud_ice = @. ᶜq.ice
         ᶜwatervapor = @. TD.vapor_specific_humidity(ᶜq)
         ᶜRH = @. TD.relative_humidity(thermo_params, ᶜts)
 
-        # precipitation
-        @. ᶜS_ρq_tot =
-            Y.c.ρ * CM.Microphysics0M.remove_precipitation(
-                cm_params,
-                TD.PhasePartition(thermo_params, ᶜts),
-            )
-
-        # rain vs snow
-        ᶜ3d_rain = @. ifelse(ᶜT >= FT(273.15), ᶜS_ρq_tot, FT(0))
-        ᶜ3d_snow = @. ifelse(ᶜT < FT(273.15), ᶜS_ρq_tot, FT(0))
-        col_integrated_rain =
-            vertical∫_col(ᶜ3d_rain) ./ FT(CAP.ρ_cloud_liq(params))
-        col_integrated_snow =
-            vertical∫_col(ᶜ3d_snow) ./ FT(CAP.ρ_cloud_liq(params))
-
         moist_diagnostic = (;
             cloud_liquid = ᶜcloud_liquid,
             cloud_ice = ᶜcloud_ice,
             water_vapor = ᶜwatervapor,
-            precipitation_removal = ᶜS_ρq_tot,
-            column_integrated_rain = col_integrated_rain,
-            column_integrated_snow = col_integrated_snow,
             relative_humidity = ᶜRH,
         )
+        if :ᶜS_ρq_tot in propertynames(Y.c)
+            # precipitation
+            @. ᶜS_ρq_tot =
+                Y.c.ρ * CM.Microphysics0M.remove_precipitation(
+                    cm_params,
+                    TD.PhasePartition(thermo_params, ᶜts),
+                )
+            # rain vs snow
+            ᶜ3d_rain = @. ifelse(ᶜT >= FT(273.15), ᶜS_ρq_tot, FT(0))
+            ᶜ3d_snow = @. ifelse(ᶜT < FT(273.15), ᶜS_ρq_tot, FT(0))
```

Comment:
These are full 3D field allocations. Let's instead add these to the cache.

Reply:
I'll fix it when I'll fix the allocations for the precipitation output. (I'm not adding those lines in this PR, just moving them behind an if statement based on the existence of precipitation sources)

Reply:
TODO for https://github.com/CliMA/ClimaAtmos.jl/issues/714

Reply:
Apologies, it's been a while. But I'll get to it once I'm done with adding some plots to TC+CA debugging output

Reply:
> I'll fix it when I'll fix the allocations for the precipitation output.

Why can't we fix it now? This is simpler than fixing #714, we just have to add `ᶜ3d_rain` and `ᶜ3d_snow` to the cache

Reply:
I thought thats what needed to be done in #714

Reply:
#714 is also an allocation issue, but it's of a different nature in that the function itself allocates. All statements of the form

```julia
x = @. y
```
are allocating, and need to be fixed.

We may be able to fix #714 in an elegant way, but these issues (fixing `x = @. y`) can be done now relatively cleanly and simply by adding these variables to the cache.